planner/optimizer/src/main/java/eu/seaclouds/platform/planner/optimizer/OptimizerInitialDeployment.java

```diff
@@ -42,69 +49,222 @@ public OptimizerInitialDeployment() {
+      } catch (Exception E) {
+         log.error("There was an exception while reading topology. More errors ahead");
+      }
       // TODO: Remove the following temporal management of the lack of topology.
       // Create an incorrect and ad-hoc one to keep the system working
       if (topology == null) {
-         log.error("Topology could not be parsed. Not known quantity of calls between modules. Assuming the dummy case where"
-               + "all modules are called in sequence. The order of calls is random}");
+         log.error(
+               "Topology could not be parsed. Not known quantity of calls between modules. Assuming the dummy case where"
+                     + "all modules are called in sequence. The order of calls is random}");
          topology = createAdHocTopologyFromSuitableOptions(appInfoSuitableOptions);
       }
 
+      // Read the quality requirements of the application
+      if (IS_DEBUG) {
+         log.info("Getting application Requirements");
+      }
+      QualityInformation requirements = loadQualityRequirements(appMap);
+      if (IS_DEBUG) {
+         log.info("following requirements found: " + requirements.toString());
+      }
+
+      // Create the skeleton of the numPlansToGenerate solutions to fill
+      // This is only to avoid using the appMap or call to the TOSCA parser
+      // inside
+
       // Compute solution
-      Map<String, Object>[] mapSolutions = engine.computeOptimizationProblem(
-            appInfoSuitableOptions.clone(), appMap, topology,
+      Solution[] solutions = engine.computeOptimizationProblem(appInfoSuitableOptions.clone(), requirements, topology,
             numPlansToGenerate);
 
-      if (mapSolutions == null) {
+      if (solutions == null) {
          log.error("Map returned by Search engine is null");
       }
 
+      Map<String, Object>[] appMapSolutions = hashMapOfFoundSolutionsWithThresholds(solutions, appMap, topology,
+            appInfoSuitableOptions, numPlansToGenerate, requirements);
+
+      if (IS_DEBUG) {
+         log.info("Before ReplaceSuitableServiceByHost");
+      }
+
       String[] stringSolutions = new String[numPlansToGenerate];
-      for (int i = 0; i < mapSolutions.length; i++) {
-         YAMLoptimizerParser.ReplaceSuitableServiceByHost(mapSolutions[i]);
-         stringSolutions[i] = YAMLoptimizerParser
-               .FromMAPtoYAMLstring(mapSolutions[i]);
+      for (int i = 0; i < appMapSolutions.length; i++) {
+         YAMLoptimizerParser.ReplaceSuitableServiceByHost(appMapSolutions[i]);
+         stringSolutions[i] = YAMLoptimizerParser.FromMAPtoYAMLstring(appMapSolutions[i]);
       }
 
       return stringSolutions;
    }
 
+   private Map<String, Object>[] hashMapOfFoundSolutionsWithThresholds(Solution[] bestSols,
+         Map<String, Object> applicMap, Topology topology, SuitableOptions cloudOffers, int numPlansToGenerate,
+         QualityInformation requirements) {
+
+      if (IS_DEBUG) {
```

Comment:
Thanks. I have used it a lot of times along. I'll change it. 
In the beggining it was with log.debug, but at some point this log.debug stopped working and I could not change the log level to show the debug messages (when we changed to testng). I did not look for a solution in that moment. I'll do it now.